ndcube/utils/wcs.py

```diff
@@ -305,16 +305,18 @@ def get_dependent_axes(wcs_object, axis):
     pc = np.array(wcs_object.wcs.get_pc()[::-1, ::-1])
     ndim = pc.shape[0]
     pc[np.eye(ndim, dtype=np.bool)] = 0
-    axes = wcs_object.get_axis_types()[::-1]
+    axes = wcs_object.get_axis_types()
+    axes = np.array(axes)[np.invert(missing_axis)][::-1]
 
-    # axes rotated
+    # axes rotated.  In a departure from where this was copied,
+    # ensure any missing axes are not returned.
     if pc[axis, :].any() or pc[:, axis].any():
-        return tuple(range(ndim))
+        return tuple(np.arange(ndim)[np.invert(missing_axis[::-1])])
 
     # XXX can spectral still couple with other axes by this point??
     if axes[axis].get('coordinate_type') != 'celestial':
         return (axis,)
 
     # in some cases, even the celestial coordinates are
     # independent. We don't catch that here.
-    return tuple(i for i, a in enumerate(axes) if a.get('coordinate_type') == 'celestial')
+    return tuple([i for i, a in enumerate(axes) if a.get('coordinate_type') == 'celestial'])
```

Comment:
why the addition of the list here?

Reply:
No reason.  Guess I never deleted them when cleaning up the code.  They can be removed.